tpke/src/lib.rs

```diff
@@ -492,16 +476,158 @@ mod tests {
 
         // Malformed the ciphertext
         ciphertext.ciphertext[0] += 1;
-        let result = std::panic::catch_unwind(|| {
+        let result = panic::catch_unwind(|| {
             checked_decrypt_with_shared_secret(&ciphertext, aad, &shared_secret)
         });
         assert!(result.is_err());
 
         // Malformed the AAD
         let aad = "bad aad".as_bytes();
-        let result = std::panic::catch_unwind(|| {
+        let result = panic::catch_unwind(|| {
             checked_decrypt_with_shared_secret(&ciphertext, aad, &shared_secret)
         });
         assert!(result.is_err());
     }
+
+    #[test]
+    fn simple_threshold_decryption_with_share_refreshing_at_point() {
+        let rng = &mut test_rng();
+        let shares_num = 16;
+        let threshold = shares_num * 2 / 3;
+
+        let (_, _, mut contexts) =
+            setup_simple::<E>(threshold, shares_num, rng);
+
+        // Prepare participants
+
+        // First, save the soon-to-be-removed participant
+        let selected_participant = contexts.pop().unwrap();
+        let x_r = selected_participant
+            .public_decryption_contexts
+            .last()
+            .unwrap()
+            .domain;
+        let original_y_r =
+            selected_participant.private_key_share.private_key_shares[0];
+
+        // Now, we have to remove the participant from the contexts and all nested structures
+        let mut remaining_participants = contexts;
+        for p in &mut remaining_participants {
+            p.public_decryption_contexts.pop();
```

Comment:
Here we're assuming that the ordering of participants is always the same. Would it be better to specifically remove `selected_participant` rather than just `pop`?